agent/tasks/executetestrun.go

```diff
@@ -104,27 +104,36 @@ func (ett ExecuteTestRunTask) Run() error {
 					log.Errorf("Failed to kill %s after timeout: %s", testletPath, err)
 				} else {
 					log.Debug("Successfully killed ", testletPath)
+
 				}
 			case err := <-done:
 				if err != nil {
 					log.Errorf("Testlet %s completed with error '%s'", testletPath, err)
-					gatheredData[thisTarget] = "error"
+
+					// TODO(mierdin): Handling testrun errors is on my plate as it is, and so
+					// this should get addressed properly in the future. The current approach
+					// of adding "error" to this map does nothing, as the status is tracked elsewhere
+					// gatheredData[thisTarget] = "error"
+
 				} else {
 					log.Debugf("Testlet %s completed without error", testletPath)
+
+					// Record test data
+					b := json.RawMessage(cmdOutput.Bytes())
+					gatheredData[thisTarget] = &b
```

Comment:
Good call. Done in https://github.com/toddproject/todd/pull/121/commits/1bcc4f177826d196b4b9d3b37a2d2516deff6208